travel/models/travel_passenger.py

```diff
@@ -20,63 +20,70 @@
-from openerp.tools.translate import _
+from openerp import fields, models, api, _
 
 
-class travel_passenger(orm.Model):
-
+class TravelPassenger(models.Model):
     """Passenger on travel"""
-    _description = _(__doc__)
+
     _name = 'travel.passenger'
+    _description = _(__doc__)
     _inherit = ['mail.thread']
     _rec_name = 'partner_id'
-    _columns = {
-        'partner_id': fields.many2one(
-            'res.partner',
-            'Passenger',
-            required=True,
-            ondelete='cascade',
-            help="Name of Passenger.",
-        ),
-        'travel_id': fields.many2one(
-            'travel.travel', 'Travel',
-            help='Travel for which the passenger is participating.'),
-        'travel_name': fields.related(
-            'travel_id', 'name', type='char', string='Travel'),
-        'travel_state': fields.related(
-            'travel_id', 'state', type='selection', string='State'),
-    }
+    partner_id = fields.Many2one(
+        'res.partner',
+        'Passenger',
+        required=True,
+        ondelete='cascade',
+        help="Name of Passenger.",
+    )
+    travel_id = fields.Many2one(
+        'travel.travel',
+        'Travel',
+        help='Travel for which the passenger is participating.'
+    )
+    travel_name = fields.Char(
+        string='Travel',
+        related='travel_id.name',
+    )
+    travel_state = fields.Selection(
+        string='State',
+        related='travel_id.state',
+    )
+
+    @api.multi
+    def name_get(self):
+        """Get name of partner"""
+        return [(i.id, i.partner_id.name_get()[0][1]) for i in self]
 
-    def name_get(self, cr, uid, ids, context=None):
-        if type(ids) is not list:
-            ids = [ids]
-        return [(i.id, i.partner_id.name_get()[0][1])
-                for i in self.browse(cr, uid, ids, context=context)]
+    @api.one
+    def action_passenger_form_view(self):
+        """Call action, if there is a travel, put it in the name.
 
-    def action_passenger_form_view(self, cr, uid, ids, context=None):
-        """Call action, if there is a travel, put it in the name."""
-        if not ids:
-            return {}
-        if type(ids) is not list:
-            ids = [ids]
-        passenger = self.browse(cr, uid, ids, context=context)[0]
-        travel = passenger.travel_id
-        travel_name = ('%s / %s ' % (travel.name, passenger.name_get()[0][1]))
+        Insert default dates in context
+        """
+        travel = self.travel_id
+        if not travel:
+            return {}  # pragma: no cover
+        travel_name = ('%s / %s ' % (travel.name, self.name_get()[0][1]))
         name = travel_name or _('Passenger')
-        context['default_date_start'] = travel.date_start
-        context['default_date_stop'] = travel.date_stop
+        context = dict(
+            self._context,
+            default_date_start=travel.date_start,
+            default_date_stop=travel.date_stop
+        )
         return {
```

Comment:
bug: this won't work: api.one will put your return value in a list, and this will not be processed as expected by the web client. You need api.multi + self.ensure_one()